tests/e2e/utils/settings.ts

```diff
@@ -2,11 +2,17 @@ import { ElectronApplication, Page } from "@playwright/test";
 import { TextDocument } from "../objects/editor/TextDocument";
 import { executeVSCodeCommand } from "./commands";
 
-const DEFAULT_UI_SETTINGS = {
-  // required for right-click context menu action to delete subject schemas
-  // (see https://code.visualstudio.com/updates/v1_101#_custom-menus-with-native-window-title-bar)
-  "window.menuStyle": "custom",
-};
+// `window.menuStyle` must be set to "custom" for right-click context menu actions (e.g. deleting
+// schemas, generating projects from sidebar resources, etc), but Windows already sets this to
+// inherit the "custom" setting value from `window.titleBarStyle`, so we shouldn't set it or it will
+// require a restart to take effect. (Also, based on our system dialog stubbing, if we updated this
+// on Windows, we would auto-reload the window and lose our Electron/page context, which will cause
+// all sorts of odd test failures.)
```

Comment:
Maybe in a future branch we adjust how we're stubbing dialogs so we don't get accidentally get into this situation again, but that's a future-me problem.